src/cil.ml

```diff
@@ -1909,27 +1876,6 @@ let getParenthLevelAttrParam (a: attrparam) =
   | AQuestion _ -> questionLevel
 
 
-(* Separate out the storage-modifier name attributes *)
-let separateStorageModifiers (al: attribute list) =
-  let isstoragemod (Attr(an, _): attribute) : bool =
-    try
-      match H.find attributeHash an with
-        AttrName issm -> issm
-      | _ -> false
-    with Not_found -> false
-  in
```

Comment:
As `attributeHash` is only filled with entry values `AttrName false`, this only returns the second list, which was the reason why you removed it for GCC.
It seems though that previously there was the attempt to handle the storage modifiers separately, but somehow it was forgotten to handle this properly?

Reply:
Are there any additional storage modifiers other than the storage class and whether something is inline? These are not part of the attributes here.

Any attributes that are associated with a type are also printed before the name of the variable as part of `self#pType`.

```
and storage =
    NoStorage                         (** The default storage. Nothing is  printed  *)
    | Static
    | Register
    | Extern
```

https://github.com/goblint/cil/blob/b6d354f69f4d1c85c78734fff32c3ed9ce917a78/src/cil.ml#L3288-L3296

So the question here is: Do we have any attributes associated with a variable that need to be printed before its name? I could not find any such case when going over our attributes, but I may very well have missed something,

Reply:
So these storage class attributes are a [MSVC specific thing](https://docs.microsoft.com/en-us/cpp/c-language/c-extended-storage-class-attributes?view=msvc-170). 

But I first misread the code that initalized the `attributeHash` hashtable, there actually were some keys that were associated with `AttrName true`. (See my other comment.) So these were previously printed before the name.